content/sensu-core/0.29/guides/troubleshooting-sensu.md

```diff
@@ -0,0 +1,336 @@
+
+If you've already gone through the steps in the previous section to confirm that your Sensu instance is using the correct credentials to connect to your RabbitMQ instance, then you'll want to proceed through this part of the guide to rule out any issues with SSL.
+
+_PRO TIP: For troubleshooting SSL issues, the openssl tool provides a wealth of troubleshooting capabilities. To see what is possible with the tool, take a look at this [handy cheat sheet][6]._
+
+#### Handshake Failures
+
+There are several layers of the proverbial onion when it comes to drilling down to handshake failures. We'll start by looking at the obvious errors that you'll see in logs, and dive deeper from there. The assumption here is that you've already configured Sensu to use SSL. If not, you'll want to refer back to our [SSL Configuration Reference material][7] before you proceed. Now, on to examining the errors you'll likely encounter in a handshake failure scenario:
+
+**Sensu Logs**:
+{{< highlight json >}}{"timestamp":"2018-06-10T16:39:15.988000+0200","level":"warn","message":"transport connection error","reason":"tcp connection lost"}
+{"timestamp":"2018-06-10T16:39:15.989000+0200","level":"warn","message":"transport connection error","reason":"possible authentication failure. wrong credentials?","user":"sensu"}{{< /highlight >}}
+
+Much like the errors seen in the previous section, the failure to connect to RabbitMQ appears to be one related to credentials. However, we can go a bit deeper by looking at the RabbitMQ logs, which present an error similar to the following:
+
+{{< highlight shell >}}2018-06-11 15:31:03.515 [info] <0.1540.0> TLS server: In state certify at ssl_handshake.erl:1289 generated SERVER ALERT: Fatal - Handshake Failure - {bad_cert,invalid_ext_key_usage}{{< /highlight >}}
+
+_NOTE: We'll presume that if you've gone through our SSL guide, that you're using the [SSL tool][8] to generate the certificates used in your deployment. If not, this is not a problem, as the commands we'll use for troubleshooting this particular scenario will prove useful no matter how your cert and key pairs are generated._
+
+Let's start off by manually verifying our certificate and key pairs. Sensu's SSL tool will place the certs/keys in the following directory:
+
+{{< highlight shell >}}
+sensu_ssl_tool
+├── client
+│ ├── cert.pem
+│ ├── keycert.p12
+│ ├── key.pem
+│ └── req.pem
+├── sensu_ca
+│ ├── cacert.cer
+│ ├── cacert.pem
+│ ├── certs
+│ │ ├── 01.pem
+│ │ └── 02.pem
+│ ├── index.txt
+│ ├── index.txt.attr
+│ ├── index.txt.attr.old
+│ ├── index.txt.old
+│ ├── openssl.cnf
+│ ├── private
+│ │ └── cakey.pem
+│ ├── serial
+│ └── serial.old
+├── server
+│ ├── cert.pem
+│ ├── keycert.p12
+│ ├── key.pem
+│ └── req.pem
+└── ssl_certs.sh{{< /highlight >}}
+
+So from that directory, we'll ensure that our server cert and key (read "cert and key that are being used inside of the RabbitMQ configuration") match
```

Comment:
Right idea here to use parentheses to maintain the connection between subject and verb. Another option would be to use match as a noun instead: "From the `sensu_ssl_tool` directory, we'll check for a match between the cert and key used inside of the RabbitMQ configuration."

Reply:
:heavy_check_mark: